model_analyzer/config/generate/model_profile_spec.py

```diff
@@ -0,0 +1,48 @@
+# Copyright (c) 2022, NVIDIA CORPORATION. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from copy import deepcopy
+from model_analyzer.config.input.objects.config_model_profile_spec import ConfigModelProfileSpec
+from model_analyzer.triton.model.model_config import ModelConfig
+
+
+class ModelProfileSpec(ConfigModelProfileSpec):
+    """
+    The profile configuration and default model config for a single model to be profiled
+    """
+
+    def __init__(self, spec: ConfigModelProfileSpec, config, client, gpus):
```

Comment:
add type checking for config, client, gpus

Reply:
done